pwiz/data/msdata/Serializer_MGF.cpp

```diff
@@ -132,6 +132,16 @@ void Serializer_MGF::Impl::write(ostream& os, const MSData& msd,
                 os << " " << intensityParam.valueFixedNotation();
             os << '\n';
 
+            CVParam inverseReduceIonMobility = scan->cvParam(MS_inverse_reduced_ion_mobility);
```

Comment:
The SpectrumListCache crash is due to a null pointer dereference here. Scan may be null so you need to handle that. It might be better to change the way Scan is defined though, like this:
`const Scan& scan = !s->scanList.empty() ? s->scanList.scans[0] : Scan();`

Then change:
`CVParam scanTimeParam = scan ? scan->cvParam(MS_scan_start_time) : CVParam();`
to
`CVParam scanTimeParam = scan.cvParam(MS_scan_start_time);`

And your new code from `scan->` to `scan.`

Reply:
Oh, I tried to find the reason. Thank you very much. I changed a lot of things, but the build still crashed. That saves me a lot of time.